lib/matplotlib/animation.py

```diff
@@ -876,6 +883,180 @@ def _args(self):
+                import warnings
+                warnings.warn('Animation size has reached {0._total_bytes} '
+                              'bytes, exceeding the limit of '
+                              '{0._bytes_limit}. If you\'re sure you want '
+                              'a larger animation embedded, set the '
+                              'animation.embed_limit rc parameter to a '
+                              'larger value (in MB). This and further frames'
+                              ' will be dropped.'.format(self))
+                self._hit_limit = True
+            else:
+                self._saved_frames.append(imgdata64)
+        else:
+            return super(HTMLWriter, self).grab_frame(**savefig_kwargs)
+
+    def _run(self):
+        # make a ducktyped subprocess standin
+        # this is called by the MovieWriter base class, but not used here.
+        class ProcessStandin(object):
+            returncode = 0
+
+            def communicate(self):
+                return ('', '')
+
+        self._proc = ProcessStandin()
+
+        # save the frames to an html file
+        if self.embed_frames:
+            fill_frames = _embedded_frames(self._saved_frames,
+                                           self.frame_format)
+        else:
+            # temp names is filled by FileMovieWriter
+            fill_frames = _included_frames(self._temp_names,
+                                           self.frame_format)
+
+        mode_dict = dict(once_checked='',
+                         loop_checked='',
+                         reflect_checked='')
+        mode_dict[self.default_mode + '_checked'] = 'checked'
+
+        interval = int(1000. / self.fps)
+
+        with open(self.outfile, 'w') as of:
+            of.write(JS_INCLUDE)
+            of.write(DISPLAY_TEMPLATE.format(id=self.new_id(),
+                                             Nframes=len(self._temp_names),
+                                             fill_frames=fill_frames,
+                                             interval=interval,
+                                             **mode_dict))
+
+
+def anim_to_jshtml(anim, fps=None, embed_frames=True, default_mode=None):
+    """Generate HTML representation of the animation"""
+    if fps is None and hasattr(anim, '_interval'):
+        # Convert interval in ms to frames per second
+        fps = 1000. / anim._interval
+
+    # If we're not given a default mode, choose one base on the value of
+    # the repeat attribute
+    if default_mode is None:
+        default_mode = 'loop' if anim.repeat else 'once'
+
+    if hasattr(anim, "_html_representation"):
+        return anim._html_representation
+    else:
+        # Can't open a second time while opened on windows. So we avoid
+        # deleting when closed, and delete manually later.
+        with tempfile.NamedTemporaryFile(suffix='.html', delete=False) as f:
+            anim.save(f.name, writer=HTMLWriter(fps=fps,
```

Comment:
Can't we just write to a BytesIO?  (when using a subprocess to encode this would be tricky due to the need to use pipes but here it should be straightforward, no?)

Reply:
I'd love to, and it makes sense, but I think that's a more invasive a change to the original code (which I didn't write) than I'm comfortable with. Animation itself currently assumes a string filename since it relies heavily on command line tools for writing.

Reply:
Alternatively, in order to make things work on Windows more easily (I have to deal with that all the time :-() you can create a TemporaryDirectory and use whatever filename you want in it -- then you still get proper cleanup at contextmanager exit.  Although that probably means depending on https://github.com/pjdelport/backports.tempfile on AntiquePython (or https://github.com/minrk/backports.temporarydirectory but that doesn't appear to be on PyPI)

Reply:
It just requires adding an in-memory option alongside the existing file option, and using that for the jshtml converter. Just a little late in the game to be adding that, especially given that I didn't write the `HTMLWriter`.